data-prepper-expression/src/main/antlr/DataPrepperExpression.g4

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+grammar DataPrepperExpression;
+
+@header {
+    package org.opensearch.dataprepper.expression.antlr;
+}
+
+fragment
+Digit
+    : '0'
+    | NonZeroDigit
+    ;
+
+fragment
+NonZeroDigit
+    : [1-9]
+    ;
+
+Integer
+    : '0'
+    | NonZeroDigit Digit*
+    ;
+
+Float
+    : NonZeroDigit? Digit '.' Digit
+    | NonZeroDigit? Digit '.' Digit+ NonZeroDigit
+    | '.' Digit
+    | '.' Digit* NonZeroDigit
+    ;
+
+Boolean
+    : TRUE
+    | FALSE
+    ;
+
+fragment
+StringCharacters
+    : StringCharacter+
+    ;
+
+fragment
+StringCharacter
+    : ~["\\$]
+    | EscapeSequence
+    ;
+
+fragment
+EscapeSequence
+    : '\\' [btnfr"'\\$]
+    ;
+
+fragment
+JsonPointerEscapeSequence
+    : '\\' [btnfr"'\\/]
+    ;
+
+fragment
+JsonPointerStringCharacters
+    : JsonPointerStringCharacter+
+    ;
+
+fragment
+JsonPointerStringCharacter
+    : ~["\\]
+    | JsonPointerEscapeSequence
+    ;
+
+fragment
+JsonPointerCharacter
+    : [A-Za-z0-9_]
+    ;
+
+fragment
+JsonPointerCharacters
+    : JsonPointerCharacter+
+    ;
+
+JsonPointer
+    : FORWARDSLASH JsonPointerCharacters? (FORWARDSLASH JsonPointerCharacters)*
+    ;
+
+EscapedJsonPointer
+    : DOUBLEQUOTE FORWARDSLASH JsonPointerStringCharacters? DOUBLEQUOTE
+    ;
+
+String
+    : DOUBLEQUOTE StringCharacters? DOUBLEQUOTE
+    ;
+
+fragment
+VariableNameLeadingCharacter
+    : [A-Za-z_]
+    ;
+
+fragment
+VariableNameCharacter
+    : VariableNameLeadingCharacter
+    | [0-9-]
+    ;
+
+fragment
+VariableNameCharacters
+    : VariableNameLeadingCharacter VariableNameCharacter*
+    ;
+
+VariableIdentifier
+    : '${' VariableNameCharacters '}'
+    ;
+
+expression
+    : binaryOperatorExpression EOF
+    | OTHER {System.err.println("unknown char: " + $OTHER.text);}
+    ;
+
+binaryOperatorExpression
+    : conditionalExpression
+    ;
```

Comment:
Is the binaryOperatorExpression redundant? It's just a simple wrapper around the conditionalExpression at this moment.

Reply:
I originally added `binaryOperatorExpression` as a convenience for an evaluator implementation. I'll remove it as it currently adds no value to the grammar and we can add it back as needed.